orchagent/cbforch.h

```diff
@@ -0,0 +1,50 @@
+#ifndef SWSS_CBFORCH_H
+#define SWSS_CBFORCH_H
+
+#include "orch.h"
+
+using namespace std;
+using namespace swss;
+
+/* Class to handle QoS map tasks. */
+class MapHandler : public unordered_map<string, sai_object_id_t>
+{
+public:
+    /* The map type handled by this handler. */
+    enum Type
+    {
+        DSCP,
+        EXP
+    };
+
+    MapHandler(Type type) : m_type(type) {}
+
+    void doTask(Consumer &consumer);
+
+private:
+    Type m_type;
+
+    /* Get the map name based on map type. */
+    const char* getMapName() const
+                        { return m_type == DSCP ? "DSCP_TO_FC" : "EXP_TO_FC"; }
+
+    sai_qos_map_list_t extractMap(const KeyOpFieldsValuesTuple &t) const;
+    sai_object_id_t createMap(const sai_qos_map_list_t &map_list);
+    bool updateMap(sai_object_id_t sai_oid,
+                   const sai_qos_map_list_t &map_list);
+    bool removeMap(sai_object_id_t sai_oid);
+};
+
+class CbfOrch : public Orch
+{
+public:
+    CbfOrch(DBConnector *db, const vector<table_name_with_pri_t> &tableNames);
+
+private:
+    MapHandler m_dscp_map;
+    MapHandler m_exp_map;
+
+    void doTask(Consumer& consumer) override;
```

Comment:
Why is this class's `doTask` method private while `MapHandler`'s is public access?  I appreciate they don't inherit from each other, but it feels like we ought to be consistent in our expectations - do we expect external classes to tell this class to do its tasks (in which case it should be public in both cases), or the method to only be internally invoked (in which case, private)?

Reply:
All the other classes inheriting from `Orch` implement the `doTask` method as private for some reason, but I can change that to be public as well, as this is how it is defined in `Orch`.